extensions/wikia/PartnerFeed/PartnerFeed.body.php

```diff
@@ -81,100 +77,105 @@ function execute() {
 		return false;
 	}
 
-
-	private function showMenu(){
-
-		global $wgOut, $wgEnableAchievementsExt, $wgEnableBlogArticles;
+	private function showMenu() {
+		$wg = F::app()->wg;
 
 		$oTmpl = new EasyTemplate( dirname( __FILE__ ) . "/templates/" );
-		$oTmpl->set_vars(
-			array (
-			    'displayBlogs' => (!empty($wgEnableBlogArticles)),
-			    'displayAchievements' => (!empty($wgEnableAchievementsExt))
-			)
-		);
-		$wgOut->addHTML( $oTmpl->render( "main-page" ) );
+		$oTmpl->set_vars( [
+			'displayBlogs' => !empty( $wg->EnableBlogArticles ),
+			'displayAchievements' => !empty( $wg->EnableAchievementsExt )
+		] );
+		$wg->Out->addHTML( $oTmpl->render( "main-page" ) );
 	}
 
-/**
- * @author Jakub Kurcek
- * @param format string 'rss' or 'atom'
- *
- */
-	private function FeedRecentBlogComments ( $format ){
-
-		global	$wgEnableBlogArticles, $wgParser, $wgUser, $wgServer,
-			$wgOut, $wgExtensionsPath, $wgRequest;
-
-		if (empty($wgEnableBlogArticles)){
+	/**
+	 * @author Jakub Kurcek
+	 *
+	 * @param string $format 'rss' or 'atom'
+	 *
+	 * @throws MWException
+	 */
+	private function FeedRecentBlogComments ( $format ) {
+		$wg = F::app()->wg;
+
+		if ( empty( $wg->EnableBlogArticles ) ) {
 			$this->showMenu();
 		} else {
-
 			// local settings
 			$maxNumberOfBlogComments = 10;
 			$userAvatarSize = 48;
 
-			$sBlogPost = $wgRequest->getText ( "blogpost", false );
-			$oTitle = Title::newFromText( $sBlogPost , 500);
-			if ( $oTitle->getArticleID() > 0 ){
+			$sBlogPost = $wg->Request->getText ( "blogpost", false );
+			$oTitle = Title::newFromText( $sBlogPost , 500 );
+			if ( $oTitle && $oTitle->getArticleID() > 0 ) {
 
-				$articleCommentList = ArticleCommentList::newFromTitle($oTitle);
+				$articleCommentList = ArticleCommentList::newFromTitle( $oTitle );
 				$articleCommentList->newFromTitle( $oTitle );
 				$aCommentPages = $articleCommentList->getCommentPages();
 
-				$counter = $maxNumberOfBlogComments;
-				$feedArray = array();
-				foreach ($aCommentPages as $commentPage){
+				$feedArray = [];
+				foreach ( $aCommentPages as $commentPage ) {
 
-					if ( ( $maxNumberOfBlogComments-- ) == 0){
+					if ( ( $maxNumberOfBlogComments-- ) == 0 ) {
 						break;
 					}
-					//make sure all data is loaded
-					if ( $commentPage[ 'level1' ]->load() ) {
-						$tmpArticleComment = $commentPage[ 'level1' ]->getData();
 
-						$feedArray[ ] = array(
+					/** @var ArticleComment $levelOne */
+					$levelOne = $commentPage['level1'];
+
+					// make sure all data is loaded
+					if ( $levelOne->load() ) {
+						$tmpArticleComment = $levelOne->getData();
+
```

Comment:
Below checks to make sure we get a revision.  If we don't get one, we log it in case there is something interesting about this situation.